### ISSUE TITLE
How to prepare the data for training

### ISSUE THREAD

**Issue by bachejames** [StevenMHernandez/Android-CSI-Labelling-App#1]
Hi there!

This is a super cool app, thanks a lot for putting it together!

I am looking to take the CSI and train it using TensorFlow. The ability to do this has been mentioned, but would love to better understand how to implement this and so have a few questions please.

1. It looks like the default state is that CSI is exported in one file (via the ESP32CSISerial library) and a separate file is created which records the training labels. Is this done purposefully? It seems like the better approach would perhaps be to first convert the CSI to amplitude via the currently unused `parseCsi` function, and then export that data whilst also appending the current label.
2. My understanding is that most training is done on just the amplitude, but if phase is also recommended, I'd love to know.
3. When training data via this method before, has any attempt been made to use CSIKit to de-noise the data or has this been done within the Android app?

Thank you so much!

**Comment by StevenMHernandez** [comment]
Thank you for your comments!

1. The goal of this is app is simply to act as a simple "labeling" or "annotation" tool. Once you have the two files annotation labels and CSI, you can bring them into your own system for preprocessing/training etc. Users are free to implement their own preprocessing in-app or offline on a server (using Tensorflow+Python), but it is not the goal of this project to give people "everything and the kitchen sink". If somebody needs very specific features, it would make sense for them to implement them on their own because they will know exactly what they want. Does that make sense?
2. Please try amplitude versus phase versus amplitude&phase. There are many options for you to try. I have primarily considered amplitude, but the data is available for you to experiment with.
3. I have not looked into CSIKit myself. Unfortunately I cannot have all of my work open source, there would be too many support questions and a lot of additional work keeping up with public documentation etc. As such, I do not currently have any public code for preprocessing the data on the Android App or offline. This is left up to the user. There are far too many different techniques for me to do this alone!

In the end, my suggestion is that you will need to implement this logic on your own because you know what your use case is and you know (or will learn) which techniques work or do not work for your use case.

I hope that helps answer your questions. Let me know if you have any more. Thanks!

**Comment by bachejames** [comment]
Very useful comments and makes a lot of sense, thank you!
